src/communication/network_interface_sr.c

```diff
@@ -5285,10 +5285,34 @@ sqmgr_execute_query (THREAD_ENTRY * thread_p, unsigned int rid, char *request, i
   trace_slow_msec = prm_get_integer_value (PRM_ID_SQL_TRACE_SLOW_MSECS);
   trace_ioreads = prm_get_integer_value (PRM_ID_SQL_TRACE_IOREADS);
 
+  base_stats = perfmon_allocate_values ();
```

Comment:
@asturzu 

I'd like to point a couple of memory management issues:

1. memory allocation on demand
2. memory leaks

The current patch always allocates memories for stats. I'd like to propose to allocate them on demand. 
Please also notice allocated memories for stats are leaked.